src/shared/clients/ec2Client.ts

```diff
@@ -19,28 +19,28 @@ export class Ec2Client {
     private async createSdkClient(): Promise<EC2> {
         return await globals.sdkClientBuilder.createAwsService(EC2, undefined, this.regionCode)
     }
-    public async getInstances(): Promise<AsyncCollection<Ec2Instance>> {
+
+    public async getInstances(filters?: EC2.Filter[]): Promise<AsyncCollection<EC2.Instance>> {
         const client = await this.createSdkClient()
+
         const requester = async (request: EC2.DescribeInstancesRequest) => client.describeInstances(request).promise()
-        const collection = pageableToCollection(requester, {}, 'NextToken', 'Reservations')
+        const collection = filters
+            ? pageableToCollection(requester, { Filters: filters }, 'NextToken', 'Reservations')
+            : pageableToCollection(requester, {}, 'NextToken', 'Reservations')
```

Comment:
syntax nit: can this be less verbose, something like (untested):

```suggestion
        const collection =pageableToCollection(requester, filters ? { Filters: filters } : {}, 'NextToken', 'Reservations')
```